File: LivrusVirtus/FoxToolKit/FXPython/ConAutores.py
from FXPy.fox import *
from Consultas import frmConsultas
from resource import Resource
from ConsultasSQL import ConsultasSQL
import Principal

class frmConAutores(frmConsultas):

  def __init__(self, owner):
    frmConsultas.__init__(self, owner, Resource.STR_CONAUTOR)  

    FXMAPFUNC(self, SEL_COMMAND, frmConAutores.ID_FRMCONSULTAS_BTNPESQUISAR, 
      frmConAutores.onBtnPesquisar)
    FXMAPFUNC(self, SEL_COMMAND, frmConAutores.ID_FRMCONSULTAS_BTNLIMPAR, 
      frmConAutores.onBtnLimpar)
    FXMAPFUNC(self, SEL_SELECTED, frmConAutores.ID_FRMCONSULTAS_GRIDCONSULTA, 
      frmConAutores.onGridSelected)
    FXMAPFUNC(self, SEL_CHANGED, frmConAutores.ID_FRMCONSULTAS_GRIDCONSULTA, 
      frmConAutores.onGridChanged)
    
    self.labDescricao.setText("Autor:")
    self.gridConsulta.setItemText(0, 2, "Autor")

  def create(self):
    frmConsultas.create(self)
  
  def onBtnPesquisar(self, sender, sel, ptr):
    self.PesquisarDados(ConsultasSQL.ConsSQL.Autor('S',"", 
      self.edtDescricao.getText()), Resource.STR_AUTNENC, 2)
    return True
  
  def onBtnLimpar(self, sender, sel, ptr):
    self.LimparDados()
    return True
  
  def onGridSelected(self, sender, sel, ptr):
    self.setLabRegistros(sender.getCurrentRow())
    return True
  
  def onGridChanged(self, sender, sel, ptr):
    self.setLabRegistros(sender.getCurrentRow())
    return True

File: LivrusVirtus/KDevelop/FOXToolKit/ODBC/livrus/livrus/ConAutores.cpp
#include "resource.h"
#include "ConAutores.h"
#include "Principal.h"
#include "ConsultasSQL.h"

CfrmConAutores *frmConAutores = (CfrmConAutores *) NULL;

FXDEFMAP(CfrmConAutores) CfrmConAutoresMap[]={
  FXMAPFUNC(SEL_COMMAND, CfrmConAutores::ID_FRMCONSULTAS_BTNPESQUISAR, 
    CfrmConAutores::onBtnPesquisar),
  FXMAPFUNC(SEL_COMMAND, CfrmConAutores::ID_FRMCONSULTAS_BTNLIMPAR, 
    CfrmConAutores::onBtnLimpar),
  FXMAPFUNC(SEL_SELECTED,CfrmConAutores::ID_FRMCONSULTAS_GRIDCONSULTA,
    CfrmConAutores::onGridSelected),
  FXMAPFUNC(SEL_CHANGED,CfrmConAutores::ID_FRMCONSULTAS_GRIDCONSULTA,
    CfrmConAutores::onGridChanged)
};
FXIMPLEMENT(CfrmConAutores,CfrmConsultas,CfrmConAutoresMap,
  ARRAYNUMBER(CfrmConAutoresMap))

CfrmConAutores::CfrmConAutores(FXWindow* owner):
  CfrmConsultas(owner, STR_CONAUTOR) {
  IniciaControles();
}

void CfrmConAutores::IniciaControles() {
  this->labDescricao->setText("Autor:");
  this->gridConsulta->setColumnText(1, "Autor");
}

void CfrmConAutores::create(){
  FXDialogBox::create();
  show(PLACEMENT_SCREEN);
}

CfrmConAutores::~CfrmConAutores(){  
}

long CfrmConAutores::onBtnPesquisar(FXObject*,FXSelector,void*) {
  PesquisarDados(
    ConsSQL->Autor('S',"", edtDescricao->getText()), STR_AUTNENC, 2);
  return true;
}

long CfrmConAutores::onBtnLimpar(FXObject*,FXSelector,void*) {
  LimparDados();
  return true;
}

long CfrmConAutores::onGridSelected(FXObject*,FXSelector,void* ptr){
  FXTableRange *escala = (FXTableRange*) ptr;
  setLabRegistros(escala->fm.row);
  return true;
}

long CfrmConAutores::onGridChanged(FXObject*,FXSelector,void* ptr){
  FXTablePos *pos = (FXTablePos*) ptr;
  setLabRegistros(pos->row);
  return true;
}
